src/dbus/methods/player_peer_info.c

```diff
@@ -34,38 +35,90 @@ void cmd_peer_player_info_dbus_cb(const char *result,
 {
     struct cb_args *a = (struct cb_args *) args;
 
+    GVariantBuilder builder;
+    GVariant *playerinfodict;
+
+    g_variant_builder_init (&builder, G_VARIANT_TYPE("a{?*}"));
+
     if (result == NULL)
     {
-      result = "-1";
+        g_variant_builder_add (&builder, "{ss}", "error", "-1");
+    }
+    else
+    {
+        g_variant_builder_add(&builder, "{ss}", "online_id", get_info(result, "online_id='", "'", NULL) ? : "");
+        g_variant_builder_add(&builder, "{ss}", "nickname", get_info(result, "nickname='", "'", NULL) ? : "");
+        g_variant_builder_add(&builder, "{ss}", "primary_weapon", get_info(result, "primary_weapon='", "'", NULL) ? : "");
+        g_variant_builder_add(&builder, "{ss}", "primary_weapon_skin", get_info(result, "primary_weapon_skin='", "'", NULL) ? : "");
+
+        g_variant_builder_add(&builder, "{ss}", "banner_badge", get_info(result, "banner_badge='", "'", NULL));
```

Comment:
Some of the fields are integers. You can use `get_info_int`

Reply:
i'm having some problem adding int to dict 
```
g_variant_builder_add(&builder, "{si}", "banner_mark", s->banner_mark);
```
this one is not working 
error:-
```
g_variant_builder_add_value: assertion '!GVSB(builder)->prev_item_type || g_variant_is_of_type (value, GVSB(builder)->prev_item_type)' failed```

Reply:
try with variants
https://developer.gnome.org/glib/stable/gvariant-format-strings.html#gvariant-format-strings-dictionaries